test-case/check-audio-equalizer.sh

```diff
@@ -68,77 +68,104 @@ current_dir=$(dirname ${BASH_SOURCE[0]})
 result=1
 failed_cnt=0
 
-# TODO: direct import only EQ pipeline and loop for EQ pipeline only
 for idx in $(seq 0 $(expr $PIPELINE_COUNT - 1))
 do
     dlogi "$idx - total pipeline= $PIPELINE_COUNT"
     dev=$(func_pipeline_parse_value $idx dev)
     channel=$(func_pipeline_parse_value $idx channel)
     rate=$(func_pipeline_parse_value $idx rate)
     fmt=$(func_pipeline_parse_value $idx fmt)
+    type=$(func_pipeline_parse_value $idx type)
     eq_support=$(func_pipeline_parse_value $idx eq)
-    if [ -z "$eq_support" ]; then
-        dlogi "None of FIR/IIF filter is available in this pipeline, skip"
-        continue
-    fi
+
+    case $type in
+        "playback")
+            cmd=aplay
+            dummy_file=/dev/zero
+        ;;
+        "capture")
+            cmd=arecord
+            dummy_file=/dev/null
+        ;;
+    esac
+
     dlogi "eq_support= $eq_support"
-    is_iir=$(echo $eq_support | grep -i iir | wc -l)
-    is_fir=$(echo $eq_support | grep -i fir | wc -l)
-    if [ $is_iir -ne $is_fir ]; then
-        dlogi "no FIR or IIF filter is available in this pipeline, skip"
-        continue
-    fi
 
-    dlogi "0. Get amixer control id for IIR and FIR"
-    IIRid=`amixer -D hw:$sofcard controls | grep EQIIR| head -1| sed 's/numid=\([0-9]*\),.*/\1/'`
-    if [ -z $IIRid ]; then
-            dloge "can't find IIR filter"
-            exit 1
-    fi
+    # if IIR filter is avilable, test wit IIR coef
+    is_iir=$(echo $eq_support | grep -i iir | wc -l)
+    if [[ $is_iir -ne "" ]]; then
+        # 1. Test IIR
+        dlogi "1.1 Get amixer control id for IIR"
+        IIRid=`amixer -D hw:$sofcard controls | grep EQIIR| head -1| sed 's/numid=\([0-9]*\),.*/\1/'`
+        if [ -z $IIRid ]; then
```

Comment:
This means `[ -z ]`, -z has no argument  if `$IIRid` is empty. Not good. Just double quote almost everything. https://mywiki.wooledge.org/Quotes